src/Centreon/Infrastructure/HostConfiguration/Repository/HostSeverityRepositoryRDB.php

```diff
@@ -0,0 +1,127 @@
+
+use Centreon\Domain\HostConfiguration\Interfaces\HostSeverityReadRepositoryInterface;
+use Centreon\Domain\RequestParameters\RequestParameters;
+use Centreon\Infrastructure\DatabaseConnection;
+use Centreon\Infrastructure\HostConfiguration\Repository\Model\HostSeverityFactoryRdb;
+use Centreon\Infrastructure\Repository\AbstractRepositoryDRB;
+use Centreon\Infrastructure\RequestParameters\Interfaces\NormalizerInterface;
+use Centreon\Infrastructure\RequestParameters\SqlRequestParametersTranslator;
+
+/**
+ * This class is designed to represent the MariaDb repository to manage host severity.
+ *
+ * @package Centreon\Infrastructure\HostConfiguration\Repository
+ */
+class HostSeverityRepositoryRDB extends AbstractRepositoryDRB implements HostSeverityReadRepositoryInterface
+{
+    /**
+     * @var SqlRequestParametersTranslator
+     */
+    private $sqlRequestTranslator;
+
+    public function __construct(DatabaseConnection $db, SqlRequestParametersTranslator $sqlRequestTranslator)
+    {
+        $this->db = $db;
+        $this->sqlRequestTranslator = $sqlRequestTranslator;
+        $this->sqlRequestTranslator
+            ->getRequestParameters()
+            ->setConcordanceStrictMode(RequestParameters::CONCORDANCE_MODE_STRICT);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function findHostSeverities(): array
+    {
+        $this->sqlRequestTranslator->setConcordanceArray(
+            [
+                'id' => 'hc_id',
+                'name' => 'hc_name',
+                'alias' => 'hc_alias',
+                'level' => 'level',
+                'icon' => 'icon_id',
+                'is_activated' => 'hc_activate',
+            ]
+        );
+        $this->sqlRequestTranslator->addNormalizer(
+            'is_activated',
+            new class implements NormalizerInterface
+            {
+                public function normalize($valueToNormalize)
+                {
+                    if (is_bool($valueToNormalize)) {
+                        return ($valueToNormalize === true) ? '1' : '0';
+                    }
+                    return $valueToNormalize;
+                }
+            }
+        );
+        $request = $this->translateDbName(
+            'SELECT SQL_CALC_FOUND_ROWS hc.*, icon.img_id AS img_id, icon.img_name AS img_name,
+                CONCAT(iconD.dir_name,\'/\',icon.img_path) AS img_path, icon.img_comment AS img_comment
+            FROM `:db`.hostcategories hc
+            LEFT JOIN `:db`.view_img icon
+                ON icon.img_id = hc.icon_id
+            LEFT JOIN `centreon`.view_img_dir_relation iconR
+                ON iconR.img_img_id = icon.img_id
+            LEFT JOIN `centreon`.view_img_dir iconD
+                ON iconD.dir_id = iconR.dir_dir_parent_id'
+        );
+
+        // Search
+        $searchRequest = $this->sqlRequestTranslator->translateSearchParameterToSql();
+        $request .= !is_null($searchRequest)
+            ? $searchRequest . ' AND level IS NOT NULL'
```

Comment:
You cannot allow the filtering of the 'level' property and then force its value in the query.
Please test all search parameters.